pkg/kv/kvserver/replica_app_batch.go

```diff
@@ -313,8 +313,12 @@ func (b *replicaAppBatch) runPostAddTriggersReplicaOnly(
 		// TODO(nvanbenschoten): It should be possible to only reject registrations
 		// that overlap with the new range of the split and keep registrations that
 		// are only interested in keys that are still on the original range running.
+		reason := kvpb.RangeFeedRetryError_REASON_RANGE_SPLIT
+		if res.Split.SplitTrigger.ManualSplit {
+			reason = kvpb.RangeFeedRetryError_REASON_MANUAL_RANGE_SPLIT
```

Comment:
Shouldn't we have added a version gate here to only use this reason once we are in 24.1? If a 23.2 node sees this retry reason, it will crash:

https://github.com/cockroachdb/cockroach/blob/7ca8340d8b1316144a9f0f53e736f168d99a0bab/pkg/kv/kvclient/kvcoord/dist_sender.go#L427

In fact, I saw this crash while running a roachtest.

Reply:
oof nice catch. lemme open a patch.